outputs.tf

```diff
@@ -1,81 +1,91 @@
-output "client_key" {
-  value = azurerm_kubernetes_cluster.main.kube_config[0].client_key
-}
-
-output "client_certificate" {
-  value = azurerm_kubernetes_cluster.main.kube_config[0].client_certificate
-}
-
-output "cluster_ca_certificate" {
-  value = azurerm_kubernetes_cluster.main.kube_config[0].cluster_ca_certificate
-}
-
-output "host" {
-  value = azurerm_kubernetes_cluster.main.kube_config[0].host
-}
-
-output "username" {
-  value = azurerm_kubernetes_cluster.main.kube_config[0].username
-}
-
-output "password" {
-  value = azurerm_kubernetes_cluster.main.kube_config[0].password
-}
-
 output "node_resource_group" {
-  value = azurerm_kubernetes_cluster.main.node_resource_group
+  value       = azurerm_kubernetes_cluster.main.node_resource_group
+  description = "The name of the Resource Group where the Kubernetes Nodes should exist."
 }
 
 output "location" {
-  value = azurerm_kubernetes_cluster.main.location
+  value       = azurerm_kubernetes_cluster.main.location
+  description = "The location where the Managed Kubernetes Cluster is created."
 }
 
 output "aks_id" {
-  value = azurerm_kubernetes_cluster.main.id
+  value       = azurerm_kubernetes_cluster.main.id
+  description = "The Kubernetes Managed Cluster ID."
 }
 
 output "kube_config_raw" {
-  sensitive = true
-  value     = azurerm_kubernetes_cluster.main.kube_config_raw
+  value       = azurerm_kubernetes_cluster.main.kube_config_raw
+  description = "Raw Kubernetes config to be used by kubectl and other compatible tools."
 }
 
 output "kube_admin_config_raw" {
-  sensitive = true
-  value     = azurerm_kubernetes_cluster.main.kube_admin_config_raw
+  value       = azurerm_kubernetes_cluster.main.kube_admin_config_raw
+  description = "Raw Kubernetes config for the admin account to be used by kubectl and other compatible tools. This is only available when Role Based Access Control with Azure Active Directory is enabled and local accounts enabled."
 }
 
 output "http_application_routing_zone_name" {
-  value = length(azurerm_kubernetes_cluster.main.addon_profile) > 0 && length(azurerm_kubernetes_cluster.main.addon_profile[0].http_application_routing) > 0 ? azurerm_kubernetes_cluster.main.addon_profile[0].http_application_routing[0].http_application_routing_zone_name : ""
+  value       = try(azurerm_kubernetes_cluster.main.http_application_routing_zone_name, null)
+  description = "The Zone Name of the HTTP Application Routing."
 }
 
-output "system_assigned_identity" {
-  value = azurerm_kubernetes_cluster.main.identity
+output "identity" {
+  value = {
```

Comment:
IMHO, returning attributes one-by-one is prune to breaking changes from the resource provider.
Maybe just returning `value = azurerm_kubernetes_cluster.main.identity[0]` is a better approach. 
WYT?